datastore_cache_api.py

```diff
@@ -0,0 +1,62 @@
+import datetime
```

Comment:
Is this file replacing one of the existing *cache_api.py files? It would be easier for me to review these files while comparing it to the existing files, but I also understand if it wasn't included in order to keep the PR size small. Same goes for the test files.

Reply:
Yes, this will replace the memcache_api.py file (and the corresponding memcache_api_test.py). Unfortunately, I need to slice it between PRs some manner and I had Greg review the other PR.